protocol/builder/produce_request_builder.go

```diff
@@ -0,0 +1,62 @@
+package builder
+
+import (
+	kafkaapi "github.com/codecrafters-io/kafka-tester/protocol/api"
+)
+
+type ProduceRequestBuilder struct {
+	correlationId   int32
+	topicData       []kafkaapi.ProduceTopicData
+	transactionalID *string
+	acks            int16
+	timeoutMs       int32
```

Comment:
Thinking a bit more about this - I think we can omit all fields we don't customize here. We do have all fields listed in the response structure itself, so the only purpose to further store them here is to allow them being customized in the builder. Let's instead only store fields that support customization (via the `With...` / `Add...` methods), and for all default values those can just be hardcoded in the `Build()` method when constructing the response/request.